weaviate/batch/crud_batch.py

```diff
@@ -552,6 +617,18 @@ def _create_data(
                     )
                     connection_count += 1
                 else:
+                    response_json = response.json()
+                    if (
+                        self._weaviate_error_retry is not None
+                        and batch_error_count < self._weaviate_error_retry.number_retries
+                    ):
+                        batch_to_retry = self._retry_on_error(response_json, data_type)
+                        if len(batch_to_retry) > 0:
+                            batch_error_count += 1
+                            batch_request = batch_to_retry
+                            continue  # run the request again, but only with objects that had errors
+
+                    self._run_callback(response_json)
```

Comment:
> Are there any cases where Weaviate returns object error and re-try would work?

Transformer timeouts, see for example [this thread from the public slack](https://weaviate.slack.com/archives/C02RRQP23K3/p1674081788162249)

> The batch creation time will have a wrong value of the last try/re-try, this will lead to a wrong estimation of the weaviate throughput for this reason:

Guess that depends on what you consider the true value :)
My main motivation for this was the transformer timeout mentioned above. And if we need to resend objects because of transformer timeouts reducing the batch size is probably a good idea.


>  (continuation) because of this the creation time would be very short and will mess up the true Weaviate throughput.
The time would be higher, eg we would send less objects the next time.

> This means that the callback is run only on the last batch_to_retry in case something failed in the batch.

Changed this, now it is called for all responses